File: problem7.py
from timeit import default_timer as timer
start_time = timer()
primes = []
num = 2
prime_count = 0
ans = -1
while True:
    flag = True
    for prime in primes:
        if num % prime == 0:
            flag = False
            break
    if flag:
        prime_count += 1
        primes.append(num)
    if prime_count == 10001:
        ans = num
        break
    num += 1

diff_time = timer()
# 104743 in 26.408278233 seconds
print("{0} in {1} seconds".format(ans, diff_time))

File: problem7.cpp
#include<iostream>
#include<vector>
#include<ctime>

using namespace std;

int main() {
	time_t begin;
	time(&begin);
	vector<int> primes;
	int count = 0;
	int i;
	for(i = 2; count < 10001; i++) {
		bool flag = true;
		for(vector<int>::iterator it = primes.begin(); it != primes.end(); it++) {
			if(i % *it == 0) {
				flag = false;
				break;
			}
		}
		if(flag) {
			primes.push_back(i);
			count++;
		}
	}
	time_t end;
	time(&end);
	double diff = difftime(end, begin);
	// 104743 in 5 seconds.
	cout<<--i<<" in "<<diff<<" seconds."; // Removing increment.
}
